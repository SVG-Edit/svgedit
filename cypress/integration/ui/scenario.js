import {
  visitAndApproveStorage
} from '../../support/ui-test-helper.js';

const testSnapshot = () => {
  cy.get('#svgcontent').cleanSnapshot();
};

describe('use various parts of svg-edit', function () {
  before(() => {
    visitAndApproveStorage();
  });

  it('check tool_source', function () {
    cy.get('#tool_source').click({force: true});
    cy.get('#svg_source_textarea')
      .type('{selectall}', {force: true})
      .type(`<svg width="640" height="480" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg">
      <g class="layer">
       <title>Layer 1</title>
        <rect id="rect" fill="#FF0000" height="70" stroke="#000000" stroke-width="5" width="94" x="69.5" y="51.5"/>
       </g>
     </svg>`, {force: true, parseSpecialCharSequences: false});
    cy.get('#tool_source_save').click({force: true});
    testSnapshot();
  });
  it('check tool_fhpath', function () {
    cy.get('#tool_fhpath')
      .click({force: true});
    cy.get('#rect')
      .trigger('mousemove', 200, 200, {force: true})
      .trigger('mousedown', 200, 200, {force: true})
      .trigger('mousemove', 20, 20, {force: true})
      .trigger('mouseup', {force: true});
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_text', function () {
    cy.get('#tool_text')
      .click({force: true});
    cy.get('#rect')
      .trigger('mousedown', 'center', {force: true})
      .trigger('mouseup', {force: true});
    // svgedit use the #text text field to capture the text
    // cy.get('#text').type('1234', {force: true});
    cy.get('#text').type('B', {force: true});
    testSnapshot();
  });

  it('check tool_clone', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_clone')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_italic', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_italic')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_bold', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_bold')
      .click({force: true});
    testSnapshot();
  });
  it('check change color', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('[data-rgb="#ffff00"]')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_anchor_start', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_anchor_start')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_anchor_middle', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_anchor_middle')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_anchor_end', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_anchor_end')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_decoration_underline', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_decoration_underline')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_decoration_overline', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_decoration_overline')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_text_decoration_linethrough', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#tool_text_decoration_linethrough')
      .click({force: true});
    testSnapshot();
  });
  it('check tool_letter_spacing', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#letter_spacing')
      .type('{selectall}', {force: true})
      .type('150', {force: true})
      .type('{enter}', {force: true});
    testSnapshot();
  });
  it('check tool_word_spacing', function () {
    cy.get('#svg_1').click({force: true});
    cy.get('#word_spacing')
      .type('{selectall}', {force: true})
      .type('10', {force: true})
      .type('{enter}', {force: true});
    testSnapshot();
  });
  it('check tool_star', function () {
    cy.get('#tool_star')
      .click({force: true});
    cy.get('#svgcontent')
      .trigger('mousedown', {which: 1, pageX: 600, pageY: 150, force: true})
      .trigger('mousemove', {which: 1, pageX: 600, pageY: 170, force: true})
      .trigger('mouseup', {force: true});
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_polygon', function () {
    cy.get('#tool_polygon')
      .click({force: true});
    cy.get('#svgcontent')
      .trigger('mousedown', {which: 1, pageX: 650, pageY: 200, force: true})
      .trigger('mousemove', {which: 1, pageX: 650, pageY: 210, force: true})
      .trigger('mouseup', {force: true});
    cy.get('#svgcontent').toMatchSnapshot();
  });
});
