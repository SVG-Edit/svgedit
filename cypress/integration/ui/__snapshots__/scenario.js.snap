exports[`use various parts of svg-edit > check tool_source #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
    ></rect>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_fhpath #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      style="pointer-events:inherit"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_clone #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_italic #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_bold #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check change color #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_start #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      style="pointer-events:inherit"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-style="italic"
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
    <polygon
      cx="407"
      cy="45"
      id="svg_3"
      shape="star"
      point="5"
      r="13.333333333333334"
      radialshift=""
      r2="5.333333333333334"
      orient="point"
      fill="#000000"
      strokecolor="#000000"
      strokeWidth="0"
      points="407,31.666666666666664 410.1348546788932,40.68524269666695 419.68075355060205,40.87977340833403 412.0723014202408,46.64809063666639 414.83713669723295,55.78689325833263 407,50.333333333333336 399.16286330276705,55.78689325833263 401.9276985797592,46.64809063666639 394.31924644939795,40.87977340833404 403.8651453211068,40.68524269666695 407,31.666666666666664 410.1348546788932,40.68524269666695 "
      stroke="#000000"
      stroke-width="0"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </polygon>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_star #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      style="pointer-events:inherit"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-style="italic"
      font-weight="bold"
      text-decoration=" underline overline line-through"
      letter-spacing="150"
      word-spacing="10"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
    <polygon
      cx="407"
      cy="45"
      id="svg_3"
      shape="star"
      point="5"
      r="13.333333333333334"
      radialshift=""
      r2="5.333333333333334"
      orient="point"
      fill="#ffff00"
      strokecolor="#000000"
      strokeWidth="0"
      points="407,31.666666666666664 410.1348546788932,40.68524269666695 419.68075355060205,40.87977340833403 412.0723014202408,46.64809063666639 414.83713669723295,55.78689325833263 407,50.333333333333336 399.16286330276705,55.78689325833263 401.9276985797592,46.64809063666639 394.31924644939795,40.87977340833404 403.8651453211068,40.68524269666695 407,31.666666666666664 410.1348546788932,40.68524269666695 "
      stroke="#000000"
      stroke-width="0"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </polygon>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_polygon #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      style="pointer-events:inherit"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-style="italic"
      font-weight="bold"
      text-decoration=" underline overline line-through"
      letter-spacing="150"
      word-spacing="10"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
    <polygon
      cx="407"
      cy="45"
      id="svg_3"
      shape="star"
      point="5"
      r="13.333333333333334"
      radialshift=""
      r2="5.333333333333334"
      orient="point"
      fill="#ffff00"
      strokecolor="#000000"
      strokeWidth="0"
      points="407,31.666666666666664 410.1348546788932,40.68524269666695 419.68075355060205,40.87977340833403 412.0723014202408,46.64809063666639 414.83713669723295,55.78689325833263 407,50.333333333333336 399.16286330276705,55.78689325833263 401.9276985797592,46.64809063666639 394.31924644939795,40.87977340833404 403.8651453211068,40.68524269666695 407,31.666666666666664 410.1348546788932,40.68524269666695 "
      stroke="#000000"
      stroke-width="0"
      style="pointer-events:inherit"
      fill-opacity="1"
      stroke-opacity="1"
    ></polygon>
    <polygon
      cx="457"
      cy="95"
      id="svg_4"
      shape="regularPoly"
      sides="5"
      orient="x"
      edge="6.666666666666667"
      fill="#ffff00"
      strokecolor="#000000"
      strokeWidth="5"
      points="462.6710053890136,95 458.7524370403971,100.39344662916632 452.4120602650961,98.33333333333333 452.4120602650961,91.66666666666667 458.7524370403971,89.60655337083368 462.6710053890136,95 "
      stroke="#000000"
      stroke-width="5"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </polygon>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_anchor_start #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="start"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_anchor_middle #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_anchor_end #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_decoration_underline #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
      text-decoration=" underline"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_decoration_overline #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
      text-decoration=" underline overline"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_text_decoration_linethrough #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
      text-decoration=" underline overline line-through"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_letter_spacing #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
      text-decoration=" underline overline line-through"
      letter-spacing="150"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;

exports[`use various parts of svg-edit > check tool_word_spacing #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer">
    <title>Layer 1</title>
    <rect
      id="rect"
      fill="#FF0000"
      height="70"
      stroke="#000000"
      stroke-width="5"
      width="94"
      x="69.5"
      y="51.5"
      fill-opacity="1"
      stroke-opacity="1"
    ></rect>
    <text
      fill="#ffff00"
      stroke="#000000"
      stroke-width="0"
      x="116"
      y="87"
      id="svg_1"
      font-size="24"
      font-family="serif"
      text-anchor="end"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      font-=""
      font-weight="bold"
      text-decoration=" underline overline line-through"
      letter-spacing="150"
      word-spacing="10"
    >
      B
    </text>
    <text
      fill="#000000"
      stroke="#000000"
      stroke-width="0"
      x="136"
      y="107"
      font-size="24"
      font-family="serif"
      text-anchor="middle"
      xml:space="preserve"
      fill-opacity="1"
      stroke-opacity="1"
      id="svg_2"
      transform="matrix(1 0 0 1 0 0)"
    >
      B
    </text>
  </g>
</svg>
`;
